realsense/plugin.cpp

```diff
@@ -35,17 +49,95 @@ class realsense : public plugin {
+            configure_camera();
+        }
+    
+    void find_supported_devices(rs2::device_list devices){
+        for (rs2::device device : devices) {
+            if (device.supports(RS2_CAMERA_INFO_PRODUCT_LINE)){
+                std::string product_line = device.get_info(RS2_CAMERA_INFO_PRODUCT_LINE); 
+                std::cout << "Found Product Line: " << product_line << std::endl;
+                if (product_line == "D400"){
+                    std::cout << "Checking for supported streams" << std::endl;
+                    std::vector<rs2::sensor> sensors = device.query_sensors();
+                    for (rs2::sensor sensor : sensors){
+                        std::vector<rs2::stream_profile> stream_profiles = sensor.get_stream_profiles();
+                        //Currently, all D4XX cameras provide infrared, RGB, and depth, so we only need to check for accel and gyro
+                        for (auto&& sp : stream_profiles)
+                        {
+                            if (sp.stream_type() == RS2_STREAM_GYRO){
+                                gyro_found = true;
+                            };
+                            if (sp.stream_type() == RS2_STREAM_ACCEL){
+                                accel_found = true;
+                            }
+                        }
+                    }
+                    if (accel_found && gyro_found){
+                        D4XXI_found = true;
+                        std::cout << "Supported D4XX found!" << std::endl;
+                    }
+                } 
+                else if (product_line == "T200"){
+                    T26X_found = true;
+                    std::cout << "T26X found! " << std::endl;
+                }
+            }
+            
+        }
+        if (!T26X_found && !D4XXI_found){
+            std::cout << "No supported Realsense device detected!" << std::endl;
+        }
+    }
+
+    void configure_camera()
+    {
+        rs2::context ctx;
+        rs2::device_list devices = ctx.query_devices();
+        find_supported_devices(devices);
+        if (realsense_cam.compare("auto") == 0) {
+            if (D4XXI_found){
+                cam_select = D4XXI;
+                std::cout << "Setting cam_select: D4XX" << std::endl;
+            }
+            else if (T26X_found){
+                cam_select = T26X;
+                std::cout << "Setting cam_select: T26X" << std::endl;
+            }
+        }
+        else if ((realsense_cam.compare("D4XX") == 0) && D4XXI_found){
+            cam_select = D4XXI;
+            std::cout << "Setting cam_select: D4XX" << std::endl;
+        }
+        else if ((realsense_cam.compare("T26X") == 0) && T26X_found){
+            cam_select = T26X;
+            std::cout << "Setting cam_select: T26X" << std::endl;
+        }
+        if (cam_select == UNSUPPORTED){
+            std::cout << "Supported device NOT found!" << std::endl;
```

Comment:
We should `abort()` here.

Reply:
Thanks, I wasn't sure what the preference was so I had this basically as a placeholder.

Reply:
Also, do we want to keep these behind a debug flag or is there a logger setup I should pass them to instead of the console?

Reply:
@e3m3, will `abort()` resolve to `std::abort()` or `ILLIXR::abort()`? I forgot...

If a device is found is found, hide it behind `#ifndef NDEBUG`. If no device is found, that should be printed unconditionally.

Reply:
> @e3m3, will abort() resolve to std::abort() or ILLIXR::abort()? I forgot...

I think neither, and the compiler should complain (ambiguous). We can stick to `ILLIXR::abort(...)`.